Initial issue seems resolved, but some different issues in the CI runs:  
```
+ git checkout external/llvm-project
+ git clean -fd external/llvm-project

Removing external/llvm-project/mlir/include/mlir/ExecutionEngine/ROCm/
Removing external/llvm-project/mlir/lib/ExecutionEngine/ROCm/


+ python3 ./mlir/utils/jenkins/static-checks/premerge-checks.py

no modified files to format
Traceback (most recent call last):
  File "./mlir/utils/jenkins/static-checks/premerge-checks.py", line 131, in <module>
    and run_clang_tidy('HEAD~1', './mlir/utils/jenkins/static-checks/clang-tidy.ignore') ):
  File "./mlir/utils/jenkins/static-checks/premerge-checks.py", line 91, in run_clang_tidy
    stdout=subprocess.PIPE, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: './clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py': './clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py'
script returned exit code 1
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,18 @@ set(MLIR_MIOPEN_DRIVER_TEST_GPU_VALIDATION 1 CACHE BOOL "Enable E2E tests with G
 set(MLIR_MIOPEN_SQLITE_ENABLED 0 CACHE BOOL "Enable SQLite integration")
 set(MLIR_ENABLE_SQLITE "${MLIR_MIOPEN_SQLITE_ENABLED}")
 
+# LLVM settings that have an effect on the MLIR dialect
+set(LLVM_TARGETS_TO_BUILD "X86;AMDGPU" CACHE STRING "")
+
+# Build the ROCm conversions and run according tests if the AMDGPU backend
+# is available
+if ("AMDGPU" IN_LIST LLVM_TARGETS_TO_BUILD)
+  set(MLIR_ROCM_CONVERSIONS_ENABLED 1)
+else()
+  set(MLIR_ROCM_CONVERSIONS_ENABLED 0)
+endif()
+add_definitions(-DMLIR_ROCM_CONVERSIONS_ENABLED=${MLIR_ROCM_CONVERSIONS_ENABLED})
+
 # Pointers to: 1) external LLVM bins/libs, and 2) MIOpen Dialect bins/libs
 set(LLVM_EXTERNAL_BIN_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/bin" CACHE PATH "")
 set(LLVM_EXTERNAL_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/llvm-project/llvm/lib" CACHE PATH "")

--- a/cmake/llvm-project.cmake
+++ b/cmake/llvm-project.cmake
@@ -7,15 +7,6 @@ set(CMAKE_BUILD_TYPE Release CACHE INTERNAL "")
 set(MLIR_CMAKE_CONFIG_DIR
    "${CMAKE_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}/cmake/mlir")
 
-# Build the ROCm conversions and run according tests if the AMDGPU backend
-# is available
-if ("AMDGPU" IN_LIST LLVM_TARGETS_TO_BUILD)
-  set(MLIR_ROCM_CONVERSIONS_ENABLED 1)
-else()
-  set(MLIR_ROCM_CONVERSIONS_ENABLED 0)
-endif()
-add_definitions(-DMLIR_ROCM_CONVERSIONS_ENABLED=${MLIR_ROCM_CONVERSIONS_ENABLED})
-
 # MLIR settings
 set(MLIR_ROCM_RUNNER_ENABLED 1 CACHE BOOL "Enable building the mlir ROCm runner")
 set(MLIR_TABLEGEN_EXE mlir-tblgen)
@@ -23,7 +14,6 @@ set(MLIR_TABLEGEN_EXE mlir-tblgen)
 # LLVM settings
 set(LLVM_ENABLE_PROJECTS "mlir;lld" CACHE STRING "List of default llvm targets")
 set(LLVM_BUILD_EXAMPLES ON CACHE BOOL "")
-set(LLVM_TARGETS_TO_BUILD "X86;AMDGPU" CACHE STRING "")
 set(LLVM_INSTALL_UTILS ON CACHE BOOL "")
 set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "")
 set(LLVM_ENABLE_ASSERTIONS ON CACHE BOOL "")

--- a/mlir/utils/jenkins/Jenkinsfile.downstream
+++ b/mlir/utils/jenkins/Jenkinsfile.downstream
@@ -53,6 +53,7 @@ pipeline {
                 cmakeBuild generator: 'Ninja',\
                     buildDir: 'build',\
                     buildType: 'Release',\
+                    cleanBuild: true,\
                     installation: 'InSearchPath',\
                     steps: [[args: 'libMLIRMIOpen']],\
                     cmakeArgs: '-DBUILD_FAT_LIBMLIRMIOPEN=ON'

--- a/mlir/utils/jenkins/Jenkinsfile.downstream-xdlop
+++ b/mlir/utils/jenkins/Jenkinsfile.downstream-xdlop
@@ -53,6 +53,7 @@ pipeline {
                 cmakeBuild generator: 'Ninja',\
                     buildDir: 'build',\
                     buildType: 'Release',\
+                    cleanBuild: true,\
                     installation: 'InSearchPath',\
                     steps: [[args: 'libMLIRMIOpen']],\
                     cmakeArgs: '-DBUILD_FAT_LIBMLIRMIOPEN=ON'

--- a/mlir/utils/jenkins/static-checks/premerge-checks.py
+++ b/mlir/utils/jenkins/static-checks/premerge-checks.py
@@ -87,7 +87,7 @@ def run_clang_tidy(base_commit, ignore_config):
     diff = remove_ignored(diff.splitlines(keepends=True), open(ignore_config, 'r'))
   else:
     ignore = pathspec.PathSpec.from_lines(pathspec.patterns.GitWildMatchPattern, [])
-  p = subprocess.Popen(['./clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py', '-p0', '-quiet'],
+  p = subprocess.Popen(['./external/llvm-project/clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py', '-p0', '-quiet'],
                        stdout=subprocess.PIPE, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
   a = ''.join(diff)
   out = p.communicate(input=a.encode())[0].decode()

